File: devicesetup/src/main/java/io/particle/android/sdk/devicesetup/model/ScanAPCommandResult.java
package io.particle.android.sdk.devicesetup.model;

import io.particle.android.sdk.devicesetup.commands.ScanApCommand;
import io.particle.android.sdk.devicesetup.commands.data.WifiSecurity;


// FIXME: this naming... is not ideal.
public class ScanAPCommandResult implements WifiNetwork {

    public final ScanApCommand.Scan scan;

    public ScanAPCommandResult(ScanApCommand.Scan scan) {
        this.scan = scan;
    }

    @Override
    public String getSsid() {
        return scan.ssid;
    }

    @Override
    public boolean isSecured() {
        return scan.wifiSecurityType != WifiSecurity.OPEN.asInt();
    }

    @Override
    public String toString() {
        return "ScanAPCommandResult{" +
                "scan=" + scan +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ScanAPCommandResult that = (ScanAPCommandResult) o;

        if (getSsid() != null ? !getSsid().equals(that.getSsid()) : that.getSsid() != null)
            return false;

        return true;
    }

    @Override
    public int hashCode() {
        return getSsid() != null ? getSsid().hashCode() : 0;
    }

}

File: devicesetup/src/main/java/io/particle/android/sdk/devicesetup/model/ScanAPCommandResult.kt
package io.particle.android.sdk.devicesetup.model

import io.particle.android.sdk.devicesetup.commands.ScanApCommand
import io.particle.android.sdk.devicesetup.commands.data.WifiSecurity
import io.particle.android.sdk.utils.SSID


// FIXME: this naming is not ideal.
class ScanAPCommandResult(val scan: ScanApCommand.Scan) : WifiNetwork {
    override val ssid: SSID? = SSID.from(scan.ssid)
    override val isSecured: Boolean
        get() = scan.wifiSecurityType != WifiSecurity.OPEN.asInt()

    override fun toString(): String {
        return "ScanAPCommandResult{" +
                "scan=" + scan +
                '}'.toString()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false

        val that = other as ScanAPCommandResult?

        return if (ssid != null) ssid == that!!.ssid else that!!.ssid == null
    }

    override fun hashCode(): Int {
        return ssid?.hashCode() ?: 0
    }

}
